### PR TITLE
Standardize edxapp env setup

### PR DESCRIPTION
**Pull request by maxrothman**
Bring edxapp in line with other IDAs for how it sets its environment
variables. This includes adding shell scripts in between supervisor conf
files and gunicorn files.

### CODE REVIEW

**Comment by maxrothman** [comment]
@edx/devops

**Comment by e0d** [comment]
This merge will need to be coordinated with https://github.com/edx/configuration/pull/2509

**Comment by maxrothman** [comment]
@e0d Do you think I should remove the tags to move this forward? #2509 will probably have to be rebased anyways.

**Comment by e0d** [comment]
I think that 2509 will land momentarily and I'd rather re-base this PR on
top of it.

Ed.

**Comment by e0d** [comment]
@maxrothman #2509 has landed.

**Comment by maxrothman** [comment]
@e0d rebased

**Comment by e0d** [comment]
:+1:

**Comment by maxrothman** [comment]
Thoughts about https://github.com/edx/configuration/pull/2512/files#diff-9bffb77ca37dcc0aead699b822b29095R639?

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -633,6 +634,27 @@ edxapp_environment:
   # yaml based configs is complete
   CONFIG_ROOT: "{{ edxapp_app_dir }}"
 
+  #NB: other apps (e.g. ecommerce) set these conditionally in *.sh.j2.
+  #In following the pattern set by how we handle settings files, I prefer this approach.
+  #Would like to discuss which we prefer before merging.
```
Comment (e0d) [inline playbooks/roles/edxapp/defaults/main.yml:642]: I'm happy with this method.

Comment (maxrothman) [inline playbooks/roles/edxapp/defaults/main.yml:642]: Then we should probably switch it in the other roles too (e.g. ecommerce).

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -78,8 +78,8 @@ EDXAPP_ELASTIC_SEARCH_CONFIG:
     - host: "localhost"
       port: 9200
 
-EDXAPP_LMS_ENV: 'lms.envs.aws'
-EDXAPP_CMS_ENV: 'cms.envs.aws'
+EDXAPP_LMS_ENV_FILE: 'lms.envs.aws'
```
Comment (feanil) [inline playbooks/roles/edxapp/defaults/main.yml]: Do we override these variables anywhere?

Whether or not we do, we should add a CHANGELOG entry.

Comment (feanil) [inline playbooks/roles/edxapp/defaults/main.yml]: Looking at how this is being used, I think we should make new variables for the file names or follow a pattern in their name like we do in other roles. ie. `lms_env` and `cms_env`.  Using the django module path as the file name appropriates a variable that had a single useful purpose into being one that is also coupled with the display name of the file on disk.  If people decide to change the name of the file on disk, they may not realize why it is preventing them from loading their LMS/CMS.

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 3]
```diff
@@ -1,12 +1,21 @@
-- name: setup the edxapp env
+- name: setup the lms env
   template: >
-    src=edxapp_env.j2 dest={{ edxapp_app_dir }}/edxapp_env
+    src=lms_env.j2 dest={{ edxapp_app_dir }}/{{ EDXAPP_LMS_ENV_FILE }}
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/deploy.yml]: Can we consolidate this with a `with_items`?

**Comment by feanil** [comment]
:+1:

**Comment by edx-webhook** [comment]
Thanks for the pull request, @maxrothman! I've created [OSPR-2101](https://openedx.atlassian.net/browse/OSPR-2101) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Pull request closed without merging**